v2/api-validator/src/server/handlers/withdrawal-handlers.ts

```diff
@@ -1,22 +1,308 @@
+import * as ErrorFactory from '../http-error-factory';
 import { FastifyReply, FastifyRequest } from 'fastify';
-import { SubAccountIdPathParam, WithdrawalCapability } from '../../client/generated';
+import { IdempotencyHandler } from '../controllers/idempotency-handler';
 import { PaginationParams, getPaginationResult } from '../controllers/pagination-controller';
-import { WITHDRAWAL_METHODS } from '../controllers/withdrawal-controller';
-import { accountsController } from '../controllers/accounts-controller';
-import * as ErrorFactory from '../http-error-factory';
+import {
+  WithdrawalController,
+  WithdrawalNotFoundError,
+} from '../controllers/withdrawal-controller';
+import {
+  BlockchainWithdrawalRequest,
+  CrossAccountWithdrawalRequest,
+  EntityIdPathParam,
+  FiatWithdrawalRequest,
+  ListOrderQueryParam,
+  SubAccountIdPathParam,
+  Withdrawal,
+  WithdrawalCapability,
+} from '../../client/generated';
+import { ControllersContainer } from '../controllers/controllers-container';
+import { assetsController } from '../controllers/assets-controller';
+
+type AccountParam = { accountId: SubAccountIdPathParam };
+
+const controllers = new ControllersContainer(
+  () => new WithdrawalController(assetsController, 5, 5)
+);
+
+/**
+ * GET Endpoints
+ */
 
 export async function getWithdrawalMethods(
-  request: FastifyRequest,
+  request: FastifyRequest<{ Querystring: PaginationParams; Params: AccountParam }>,
```

Comment:
Define separate types for all the query and param properties, the way I did in trading. In all the handlers. For all the common params there are already types in `request-types.ts`.